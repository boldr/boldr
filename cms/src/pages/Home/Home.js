import React from 'react';
import Helmet from 'react-helmet';
import { Grid, Row, Col, Hero, Heading, PrimaryHeader } from 'components';

const Home = () => {
  return (
    <div>
      <Helmet title="Home" />
        <PrimaryHeader />
        <Hero />
        <Grid fluid>
        <Row>
          <Heading size={ 1 }>Why shouldn't your CMS be a little, Boldr?</Heading>
        </Row>
      </Grid>
    </div>
  );
};

export default Home;
