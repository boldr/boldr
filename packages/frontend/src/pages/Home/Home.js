/* @flow */

import React from 'react';
import Helmet from 'react-helmet';
import styled from 'styled-components';
// internal
import Heading from '@boldr/ui/Heading';
import Paragraph from '@boldr/ui/Paragraph';
import Loader from '@boldr/ui/Loader';
import { ArticleCard } from '../../scenes/Blog/components';
import type { ArticlesType } from '../../types/boldr';

const HomeHero = styled.div`
  position: relative;
  background-color: #261d16;
  background-image: url("${props => props.bgsrc}");
  background-size: cover;
  height: 400px;
  overflow: hidden;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  text-align: center;
  padding: 0 10px;
  color: white;
  h1,
  p {
    z-index: 2;
  }
`;

const HomeText = 'Meet Boldr.';
const Container = styled.div`
  flex: 1;
  max-width: 1034px;
  width: 100%;
  margin: 30px auto 100px;
  display: flex;
  flex-wrap: wrap;
`;
const CoverShadow = styled.div`
  position: absolute;
  top: 0;
  left: 0;
  right: 0;
  height: 200px;
  background-image: linear-gradient(0, rgba(0, 0, 0, 0), rgba(0, 0, 0, 0.6));
  z-index: 0;
`;
type Props = {
  articles: ArticlesType,
  isLoading: boolean,
  error?: Object,
};

class Home extends React.Component<Props, *> {
  renderArticles = () => {
    const { articles } = this.props;
    return articles.map(article => <ArticleCard key={article.id} {...article} />);
  };
  render() {
    const { isLoading } = this.props;
    return (
      <div>
        <Helmet title="Home" />
        <HomeHero bgsrc="http://i.magaimg.net/img/1f5w.jpg">
          <CoverShadow />
          <Heading type="h1" text={HomeText} isLight />
        </HomeHero>
        <Container>{isLoading ? <Loader /> : this.renderArticles()}</Container>
      </div>
    );
  }
}

export default Home;
