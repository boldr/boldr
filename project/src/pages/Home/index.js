// @flow
import React from 'react';
import { graphql } from 'react-apollo';
import universal from 'react-universal-component';
import ARTICLES_QUERY from '../../scenes/Blog/gql/articles.graphql';
import type { ArticlesType } from '../../types/boldr';

const UniversalHome = universal(import('./Home'));

type Props = {
  loading: boolean,
  error?: Object,
  articles: ArticlesType,
};

const Home = ({ loading, error, articles }: Props) => (
  <UniversalHome isLoading={loading} error={error} articles={articles} />
);
// $FlowIssue
export default graphql(ARTICLES_QUERY, {
  // $FlowIssue
  options: () => ({
    variables: {
      offset: 0,
      limit: 20,
    },
  }),
  props: ({ data: { loading, error, articles } }) => ({
    loading,
    error,
    articles,
  }),
})(Home);
