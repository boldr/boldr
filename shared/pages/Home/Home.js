/* @flow */

import React from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';

import BaseTemplate from '../templates/Base';
import { Grid, Row, Hero, Footer } from '../../components/index';

const Home = () => {
  return (
    <div>
      <BaseTemplate
        helmetMeta={ <Helmet title="Home" /> }
        hero={ <Hero /> }
        footer={ <Footer /> }
      >
      <Grid fluid>
        <Row style={ { padding: '25px' } }>
          Placeholder
        </Row>
      </Grid>
      </BaseTemplate>
    </div>
  );
};


export default Home;
