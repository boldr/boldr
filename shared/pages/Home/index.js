/* @flow */

export { default } from './Home';

/*
import { getAsyncInjectors, loadRoute, errorLoading } from '../../core/utils';

export default {
  path: '/',
  getComponent(nextState: Object, cb: Function) {
    System.import('./Home')
      .then(loadRoute(cb))
      .catch(errorLoading);
  },
};

*/
