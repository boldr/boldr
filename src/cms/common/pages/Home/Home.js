/* @flow */

import React from 'react';
import { connect } from 'react-redux';
import Helmet from 'react-helmet';
import { Header, Icon, Segment, Container } from 'semantic-ui-react';
import { Row, Hero, PrimaryHeader, Footer } from 'components/index';
import { fetchPagesIfNeeded } from 'state/index';
import PostListing from 'scenes/Blog/PostListing';
import PageTemplate from '../../theme/Boldr/PageTemplate';

type Props = {
  loaded: Boolean,
  pages: Object,
  entities: Object,
  dispatch: Function
};

const Home = (props: Props) => {
  return (
      <PageTemplate
        helmetMeta={ <Helmet title="Home" /> }
        header={ <PrimaryHeader /> }
        hero={ <Hero /> }
        footer={ <Footer /> }
      >
      <Container fluid>
        <Row style={ { padding: '25px' } }>
        <Segment>
          <Header as="h1">
            <Icon name="newspaper" />
            <Header.Content>
              Recent posts
            </Header.Content>
          </Header>
          </Segment>
        </Row>
        <PostListing />
      </Container>
      </PageTemplate>
  );
};

const mapStateToProps = (state) => {
  return {
    pages: state.boldr.pages,
    loaded: state.boldr.pages.loaded,
  };
};

export default connect(mapStateToProps, { fetchPagesIfNeeded })(Home);
