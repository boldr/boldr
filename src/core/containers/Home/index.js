import HomeContainer from 'core/containers/Home/Home.jsx';
export default HomeContainer;
