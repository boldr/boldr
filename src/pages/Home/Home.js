import React from 'react';
import Helmet from 'react-helmet';

import { Grid, Row, Col, Hero, Heading, Header } from 'components';

const Home = () => {
  return (
    <div>
      <Helmet title="Home" />
        <Header theme="dark" />
        <Hero />
        <Grid fluid>
        <Row>
          <Heading size={ 1 }>Welcome to the Home. Stay tuned...</Heading>
        </Row>
      </Grid>
    </div>
  );
};

export default Home;
