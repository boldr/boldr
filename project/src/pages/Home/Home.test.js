import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Home from './Home';

const articles = [{ id: 1, title: 'test' }, { id: 2, title: 'Testing' }];
describe('<Home />', () => {
  it('renders <Home /> without breaking', () => {
    const wrapper = shallow(<Home articles={articles} isLoading={false} />);
    expect(wrapper.instance().props.articles).toEqual(articles);
  });
  it('renders snapshot', () => {
    const wrapper = shallow(<Home articles={articles} isLoading={false} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
