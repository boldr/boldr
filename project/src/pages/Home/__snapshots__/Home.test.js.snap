// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> renders snapshot 1`] = `
<Connect(Apollo(ReactComponent))>
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
    title="Home"
  />
  <Home__HomeHero
    bgsrc="http://i.magaimg.net/img/1f5w.jpg"
  >
    <Home__CoverShadow />
    <Heading
      isLight={true}
      text="Meet Boldr."
      type="h1"
    />
  </Home__HomeHero>
  <Home__Container>
    <ArticleCard
      id={1}
      key="1"
      title="test"
    />
    <ArticleCard
      id={2}
      key="2"
      title="Testing"
    />
  </Home__Container>
</Connect(Apollo(ReactComponent))>
`;
